packaging/setup/plugins/ovirt-engine-common/ovirt-engine-keycloak/core/admin_user.py

```diff
@@ -0,0 +1,85 @@
+
+# ovirt-engine-setup -- ovirt engine setup
+#
+# Copyright oVirt Authors
+# SPDX-License-Identifier: Apache-2.0
+#
+#
+
+"""Ovirt engine keycloak initial admin user plugin."""
+
+import datetime
+import gettext
+import os
+
+from otopi import plugin
+from otopi import util
+
+from ovirt_engine_setup import constants as osetupcons
+from ovirt_engine_setup import util as osetuputil
+from ovirt_engine_setup.engine import constants as oenginecons
+from ovirt_engine_setup.engine_common import constants as oengcommcons
+from ovirt_engine_setup.keycloak import constants as okkcons
+
+
+def _(m):
+    return gettext.dgettext(message=m, domain='ovirt-engine-keycloak')
+
+
+@util.export
+class Plugin(plugin.PluginBase):
+    """Ovirt engine keycloak initial admin user plugin."""
+
+    def __init__(self, context):
+        super(Plugin, self).__init__(context=context)
+
+    @plugin.event(
+        stage=plugin.Stages.STAGE_MISC,
+        condition=lambda self: (
+                self.environment[
+                    oenginecons.CoreEnv.ENABLE
+                ] and
+                self.environment[
+                    oenginecons.EngineDBEnv.NEW_DATABASE
+                ]
+
+        ),
+    )
+    def _create_admin(self):
+        password = self.environment.get(oenginecons.ConfigEnv.ADMIN_PASSWORD)
+        if password:
+            self.logger.info(_('Creating initial Keycloak admin user'))
```

Comment:
Let me first provide more details about the process. In short it is idempotent. 
'Add user script'  creates only a json file that is later transfered to Wildfly runtime directory from where keycloak app consumes it (it's automatically removed once properly processed).  The processing takes part only when keycloak app is being deployed on Widlfly (on startup).
In order to let a user to re-use the same admin user ie.in case of manual intervention, the original copy from Widlfly server config directory (--sc) is left behind. On any subsequent invocations that copy is renamed with timestamp suffix so there is no conflict. 

With all those details do you still think it makes sense to introduce transactions? 
I'm not sure it that's technically possible but could I include that newly created json in transaction after it was created?  This way perhaps that manual rename would not be necessary.

Reply:
As I wrote, if you are certain this is safe as-is, ok - just add a comment. Re whether it's possible - yes, in principle - if you add your own transactionElement, you can do whatever you want there - it has a very simple API. You can e.g. do all you do here in its 'prepare', nothing in 'commit', and a rename/remove/whatever in 'abort'. This will work more-or-less exactly like now, other than automatic 'abort' on transaction rollback.

Reply:
For now I left a comment about TransactionElement plus information why current implementation is idempotent.